tuplex/core/src/logical/UDFOperator.cc

```diff
@@ -63,57 +64,68 @@ namespace tuplex {
-                                                                     " or alter UDF to not yield exceptions only.");
-                            // @TODO: print nice traceback with sample...
-                            // => should use beefed up sample processor class for this...
+                            throw std::runtime_error("TODO: implement via sampling of Python UDF executed most likely type, so pipeline can still get compiled.");
                             return Schema::UNKNOWN;
                         } else {
-                            // all good, keep sampled type but mark as non compilable
+                            // check what return type was given
+                            auto output_type = _udf.getOutputSchema().getRowType();
+                            if(output_type.isExceptionType()) {
+                                _udf.markAsNonCompilable();
+                                // exception type? => sample produced only exceptions! warn user.
+                                Logger::instance().defaultLogger().error("Tracing via sample produced mostly"
+                                                                         " exceptions, please increase sample size"
+                                                                         " or alter UDF to not yield exceptions only.");
+                                // @TODO: print nice traceback with sample...
+                                // => should use beefed up sample processor class for this...
+                                return Schema::UNKNOWN;
+                            } else {
+                                // all good, keep sampled type but mark as non compilable
+                            }
                         }
                     }
                 }
             }
 
-            return _udf.getOutputSchema();
-        } else {
+            if(_udf.getTypeError().empty() || _udf.getReturnTypeError() != CompileError::TYPE_ERROR_NONE) {
+                // if unsupported types presented, use sample to determine type and use fallback mode (except for list return type error, only print error messages for now)
+                return _udf.getOutputSchema();
+            }
 
-            // @Todo: support here dict syntax...
-            // @TODO: this feels redundant, i.e. tracerecord visitor should have already dealt with this...
+            // unsupported type exists, print warning
+            _udf.markAsNonCompilable();
+            for (const auto& err : _udf.getTypeError()) {
+                Logger::instance().defaultLogger().error(_udf.compileErrorToStr(err));
+            }
+            Logger::instance().defaultLogger().error("will use fallback mode");
+        }
 
-            auto pickledCode = _udf.getPickledCode();
-            auto processed_sample = getSample(typeDetectionSampleSize);
+        // @Todo: support here dict syntax...
```

Comment:
what are these comments referring to? (it seems like they're older, can we remove them?)

Reply:
They are old comments. Should we remove them? @LeonhardFS

Reply:
yes, let's remove old comments.